resharper/resharper-unity/src/HlslSupport/ShaderLabInclusionContextProvider.cs

```diff
@@ -30,11 +31,14 @@ public class ShaderLabInclusionContextProvider : IInclusionContextProvider
                 LanguageKind = CppLanguageKind.HLSL, ClrSupport = VCXCompileAsManagedOptions.ManagedNotSet,
             };
 
+            properties.PredefinedMacros.Add(CppPPDefineSymbol.ParsePredefinedMacro("SHADER_API_D3D11"));
             var cgIncludeFolder =
                 CgIncludeDirectoryTracker.GetCgIncludeFolderPath(cache.Solution.GetComponent<UnityVersion>());
             if (!cgIncludeFolder.IsEmpty)
             {
                 properties.IncludePaths.Add(cgIncludeFolder);
+                properties.IncludePaths.Add(cache.Solution.SolutionDirectory);
+                properties.IncludePaths.Add(cache.Solution.SolutionDirectory.Combine("Library"));
```

Comment:
Will adding solution directory handle includes starting with `Assets/`?

Is there anything under `Library` that can be included, or is that just for `Library/PackageCache`?

Reply:
It is just for package cache, we could remove PackageCache and add to search path Library/PackageCache and Packages folder, but then we will need to add unity specific logic for determing priority of that folders to cpp logic, thus I decide to add Library as seperated include source folder